### PR TITLE
[camera][Android] Migrate the camera view module to new API

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

A follow-up to the https://github.com/expo/expo/pull/18703.
Uses a new API in the camera module.

# How

Replaced the existing view manager and module with a new implementation that is written using the sweet API. The underlying view and logic remained almost unchanged.

Needs to work
- https://github.com/expo/expo/pull/18774
- https://github.com/expo/expo/pull/18779
- https://github.com/expo/expo/pull/18777
- https://github.com/expo/expo/pull/18776
- https://github.com/expo/expo/pull/18773

# Test Plan

- test-suite ✅
- NCL ✅

### CODE REVIEW

**Review by Kudo — APPROVED** [review]
👏

File: packages/expo-camera/android/src/main/java/expo/modules/camera/CameraModule.kt [hunk 1]
```diff
@@ -2,216 +2,185 @@ package expo.modules.camera
...
 
-import java.lang.Exception
+    AsyncFunction("pausePreview") { viewTag: Int ->
+      val view = findView(viewTag)
 
-class CameraModule(
-  context: Context,
-  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate()
-) : ExportedModule(context) {
+      if (view.isCameraOpened) {
```
Comment (tsapeta) [inline packages/expo-camera/android/src/main/java/expo/modules/camera/CameraModule.kt:54]: No need to change it in this PR if that's the old property, but it should be `isCameraOpen` 😄 basically "opened" is correct only when used as a verb

File: packages/expo-camera/android/src/main/java/expo/modules/camera/CameraViewManager.kt [hunk 2]
```diff
@@ -1,116 +1,95 @@
...
+        "onMountError",
+        "onBarCodeScanned",
+        "onFacesDetected",
+        "onFaceDetectionError",
+        "onPictureSaved"
+      )
+
+      OnViewDestroys<ExpoCameraView> { view ->
+        val uIManager = appContext.legacyModule<UIManager>()
+        uIManager?.unregisterLifecycleEventListener(view)
```
Comment (tsapeta) [inline packages/expo-camera/android/src/main/java/expo/modules/camera/CameraViewManager.kt:31]:
```suggestion
        val uiManager = appContext.legacyModule<UIManager>()
        uiManager?.unregisterLifecycleEventListener(view)
```

File: packages/expo-camera/android/src/main/java/expo/modules/camera/CameraViewManager.kt [hunk 3]
```diff
@@ -1,116 +1,95 @@
...
+
+      Prop("whiteBalance") { view: ExpoCameraView, whiteBalance: Int ->
+        view.whiteBalance = whiteBalance
+      }
+
+      Prop("pictureSize") { view: ExpoCameraView, size: String? ->
+        if (size == null) {
+          return@Prop
+        }
+
```
Comment (tsapeta) [inline packages/expo-camera/android/src/main/java/expo/modules/camera/CameraViewManager.kt:70]:
```suggestion
```

**Review by tsapeta — APPROVED** [review]

**Comment by wschurman** [comment]
I believe this is causing compilation errors for the home app (expo go), namely in `HomeActivity` and `ExperiencePackagePicker`.

To repro:
1. Open android folder in android studio
2. Build

**Comment by wschurman** [comment]
Fix: https://github.com/expo/expo/pull/18811

**Pull request merged**